<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Sydney Precision Bioinformatics Group" />

<meta name="date" content="2019-11-04" />

<title>CPC workshop</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">SingleCellPlus - CPC</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="qc.html">QC Process</a>
</li>
<li>
  <a href="scMerge.html">scMerge</a>
</li>
<li>
  <a href="downstream.html">Downstream Analysis</a>
</li>
<li>
  <a href="acknowledgement.html">Acknowledgement</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">CPC workshop</h1>
<h4 class="author">Sydney Precision Bioinformatics Group</h4>
<h4 class="date">04 November 2019</h4>

</div>


<div id="single-cell-plus" class="section level1">
<h1>Single Cell Plus</h1>
<p>This workshop is part of a joint research initiative between Hong Kong University and the University of Sydney.</p>
</div>
<div id="aim-of-this-workshop" class="section level1">
<h1>Aim of this workshop</h1>
<p>In this workshop we will focus on two mouse liver datasets to illustrate three critical topics in single-cell RNA-Seq analysis.</p>
<ol style="list-style-type: decimal">
<li>Quality control of sc-RNASeq data,</li>
<li>The <code>scMerge</code> package for integrating multiple sc-RNASeq data,</li>
<li>Some possible downstream analysis after <code>scMerge</code>.</li>
</ol>
<p>At the end of this workshop you should have some basic understanding of scRNASeq data and some key computational and statistical challenges of this type of data.</p>
</div>
<div id="prerequisites" class="section level1">
<h1>Prerequisites</h1>
<p>Ideally, you should be somewhat familiar with <code>R</code>. However, if you haven’t used <code>R</code> before, don’t worry! This is not a programming workshop, you can still pick up important practical skills by running the codes we provided.</p>
<div id="running-workshop-on-the-cloud" class="section level2">
<h2>Running workshop on the cloud</h2>
<p>Your instructor will provide you with the link to a Google Cloud server, username and the password. Once you log in, you will have access to RStudio via a webpage.</p>
<p>Type the following code into your console and you are good to go!</p>
<pre><code>source(&quot;/home/user_setup.R&quot;)</code></pre>
</div>
<div id="running-workshop-locally" class="section level2">
<h2>Running workshop locally</h2>
<p>Please try to download and install:</p>
<ul>
<li>Latest <code>R</code> from <a href="https://cloud.r-project.org/" class="uri">https://cloud.r-project.org/</a></li>
<li>Latest <code>RStudio</code> from <a href="https://www.rstudio.com/products/rstudio/download/" class="uri">https://www.rstudio.com/products/rstudio/download/</a></li>
<li>Download the data from <a href="https://storage.googleapis.com/scp_data/data.zip" class="uri">https://storage.googleapis.com/scp_data/data.zip</a></li>
<li>Install all the necessary packages by running <a href="https://raw.githubusercontent.com/SydneyBioX/SC_CPC_workshop/master/install.R">this script</a></li>
</ul>
</div>
</div>
<div id="clipboard" class="section level1">
<h1>Clipboard</h1>
<p>In some circumstances, we might need to share codes that are not currently a part of the existing materials. Please click <a href="https://docs.google.com/document/d/136YffzxjkUM8RIGsnjD_yOkCTWw4Ijdwjy4_58nC-zc/edit?usp=sharing">here</a> to access these codes.</p>
</div>
<div id="references" class="section level1">
<h1>References</h1>
<p><strong>Methodologies</strong></p>
<ol style="list-style-type: decimal">
<li><p><a href="http://www.pnas.org/lookup/doi/10.1073/pnas.1820006116">scMerge leverages factor analysis, stable expression, and pseudoreplication to merge multiple single-cell RNA-seq datasets</a>, Proceedings of the National Academy of Sciences, United States of America, 2019. Yingxin Lin, Shila Ghazanfar, Kevin Y.X. Wang, Johann A. Gagnon-Bartsch, Kitty K. Lo, Xianbin Su, Ze-Guang Han, John T. Ormerod, Terence P. Speed, Pengyi Yang, Jean Y. H. Yang.</p></li>
<li><p><a href="https://www.biorxiv.org/content/biorxiv/early/2018/11/22/229815.full.pdf">Evaluating stably expressed genes in single cells</a> (Under review). Yingxin Lin, Shila Ghazanfar, Dario Strbenac, Andy Wang, Ellis Patrick, Dave Lin, Terence Speed, Jean Y. H. Yang, Pengyi Yang.</p></li>
<li><p><a href="https://doi.org/10.1093/bib/bby076">Impact of similarity metrics on single-cell RNA-seq data clustering</a>, Briefings in Bioinformatics, 2018. Taiyun Kim, Irene Rui Chen, Yingxin Lin, Andy Y. Y. Wang, Jean Y. H. Yang, Pengyi Yang.</p></li>
</ol>
<p><strong>Data</strong></p>
<ol style="list-style-type: decimal">
<li><p><a href="https://doi.org/10.1186/s12864-017-4342-x">Single-cell RNA-Seq analysis reveals dynamic trajectories during mouse liver development</a>, BMC Genomics, 2017. Xianbin Su, Yi Shi, Xin Zou, Zhao-Ning Lu, Gangcai Xie, Jean Y. H. Yang, Chong-Chao Wu, Xiao-Fang Cui, Kun-Yan He, Qing Luo, Yu-Lan Qu, Na Wang, Lan Wang, Ze-Guang Han.</p></li>
<li><p><a href="https://doi.org/10.1002/hep.29353">A single‐cell transcriptomic analysis reveals precise pathways and regulatory mechanisms underlying hepatoblast differentiation</a>, Hepatology, 2017. Li Yang, Wei‐Hua Wang, Wei‐Lin Qi, Zhen Guo, Erfei Bi, Cheng‐Ran Xu.</p></li>
</ol>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>## R version 3.6.1 (2019-07-05)
## Platform: x86_64-apple-darwin15.6.0 (64-bit)
## Running under: macOS High Sierra 10.13.6
## 
## Matrix products: default
## BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
## LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] en_AU.UTF-8/en_AU.UTF-8/en_AU.UTF-8/C/en_AU.UTF-8/en_AU.UTF-8
## 
## attached base packages:
##  [1] splines   parallel  stats4    stats     graphics  grDevices utils    
##  [8] datasets  methods   base     
## 
## other attached packages:
##  [1] monocle_2.13.0              DDRTree_0.1.5              
##  [3] irlba_2.3.3                 VGAM_1.1-1                 
##  [5] Matrix_1.2-17               plyr_1.8.4                 
##  [7] ggpubr_0.2.3                magrittr_1.5               
##  [9] viridis_0.5.1               viridisLite_0.3.0          
## [11] MAST_1.11.4                 ggplot2_3.2.1              
## [13] cluster_2.1.0               Rtsne_0.15                 
## [15] mclust_5.4.5                scdney_0.1.5               
## [17] edgeR_3.27.13               limma_3.41.18              
## [19] dplyr_0.8.3                 SingleCellExperiment_1.7.11
## [21] SummarizedExperiment_1.15.9 DelayedArray_0.11.8        
## [23] BiocParallel_1.19.4         matrixStats_0.55.0         
## [25] Biobase_2.45.1              GenomicRanges_1.37.17      
## [27] GenomeInfoDb_1.21.2         IRanges_2.19.17            
## [29] S4Vectors_0.23.25           BiocGenerics_0.31.6        
## 
## loaded via a namespace (and not attached):
##   [1] snow_0.4-3             backports_1.1.5        Hmisc_4.2-0           
##   [4] igraph_1.2.4.1         lazyeval_0.2.2         densityClust_0.3      
##   [7] fastICA_1.2-2          amap_0.8-17            digest_0.6.21         
##  [10] foreach_1.4.7          htmltools_0.4.0        checkmate_1.9.4       
##  [13] doParallel_1.0.15      mixtools_1.1.0         recipes_0.1.7         
##  [16] graphlayouts_0.5.0     gower_0.2.1            docopt_0.6.1          
##  [19] colorspace_1.4-1       ggrepel_0.8.1          pan_1.6               
##  [22] xfun_0.10              sparsesvd_0.2          crayon_1.3.4          
##  [25] RCurl_1.95-4.12        lme4_1.1-21            zeallot_0.1.0         
##  [28] survival_2.44-1.1      iterators_1.0.12       glue_1.3.1            
##  [31] polyclip_1.10-0        gtable_0.3.0           ipred_0.9-9           
##  [34] zlibbioc_1.31.0        XVector_0.25.0         jomo_2.6-9            
##  [37] abind_1.4-5            scales_1.0.0           pheatmap_1.0.12       
##  [40] mvtnorm_1.0-11         Rcpp_1.0.2             htmlTable_1.13.2      
##  [43] proxy_0.4-23           foreign_0.8-72         Formula_1.2-3         
##  [46] lava_1.6.6             prodlim_2018.04.18     htmlwidgets_1.5.1     
##  [49] FNN_1.1.3              RColorBrewer_1.1-2     ellipsis_0.3.0        
##  [52] acepack_1.4.1          mice_3.6.0             pkgconfig_2.0.3       
##  [55] farver_1.1.0           nnet_7.3-12            locfit_1.5-9.1        
##  [58] caret_6.0-84           labeling_0.3           tidyselect_0.2.5      
##  [61] rlang_0.4.0            reshape2_1.4.3         pbmcapply_1.5.0       
##  [64] munsell_0.5.0          tools_3.6.1            generics_0.0.2        
##  [67] broom_0.5.2            ggridges_0.5.1         evaluate_0.14         
##  [70] stringr_1.4.0          yaml_2.2.0             ModelMetrics_1.2.2    
##  [73] knitr_1.25             tidygraph_1.1.2        RANN_2.6.1            
##  [76] randomForest_4.6-14    purrr_0.3.2            mitml_0.3-7           
##  [79] dendextend_1.12.0      ggraph_2.0.0           nlme_3.1-141          
##  [82] slam_0.1-45            compiler_3.6.1         rstudioapi_0.10       
##  [85] ggsignif_0.6.0         e1071_1.7-2            tibble_2.1.3          
##  [88] tweenr_1.0.1           DescTools_0.99.29      stringi_1.4.3         
##  [91] lattice_0.20-38        hopach_2.45.0          nloptr_1.2.1          
##  [94] HSMMSingleCell_1.5.0   vctrs_0.2.0            pillar_1.4.2          
##  [97] lifecycle_0.1.0        combinat_0.0-8         cowplot_1.0.0         
## [100] data.table_1.12.4      bitops_1.0-6           R6_2.4.0              
## [103] latticeExtra_0.6-28    gridExtra_2.3          codetools_0.2-16      
## [106] boot_1.3-23            MASS_7.3-51.4          assertthat_0.2.1      
## [109] minpack.lm_1.2-1       withr_2.1.2            qlcMatrix_0.9.7       
## [112] GenomeInfoDbData_1.2.1 expm_0.999-4           doSNOW_1.0.18         
## [115] grid_3.6.1             rpart_4.1-15           timeDate_3043.102     
## [118] tidyr_1.0.0            class_7.3-15           minqa_1.2.4           
## [121] rmarkdown_1.16         segmented_1.0-0        clusteval_0.1         
## [124] ggforce_0.3.1          lubridate_1.7.4        base64enc_0.1-3</code></pre>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
