---
title: "HKU workshop"
author: "Sydney Precision Bioinformatics Group"
date: "18 June 2019"
output: html_document
---

# The Single Cell Plus Workshop

This workshop is part of a joint research initiative between Hong Kong University and the University of Sydney.

# Update on August 2019

We will move this workshop to Google Cloud in the future. Some materials needs to be re-written.

# Aim of this workshop

In this workshop we will focus on two mouse liver datasets to illustrate three critical topics in single-cell RNA-Seq analysis.

  1. Quality control of sc-RNASeq data,
  1. The `scMerge` package for integrating multiple sc-RNASeq data, 
  1. Some possible downstream analysis after `scMerge`. 


At the end of this workshop you should have some basic understanding of scRNASeq data and some key computational and statistical challenges of this type of data. 


# Prerequisites

Ideally, you should be somewhat familiar with `R`. However, if you haven't used `R` before, don't worry! This is not a programming workshop, you can still pick up important practical skills by running the codes we provided. 

Please try to download and install:

  + `R 3.6` from https://cloud.r-project.org/ 
  + `RStudio` from https://www.rstudio.com/products/rstudio/download/



# Installation 

You can find all the data and materials [here](http://shiny.maths.usyd.edu.au/singlecellplus/SingleCellPlus_zip.zip). 

You should be able to run the codes below in `R` to install all the packages needed in this workshop. If you encounter any problems. Please let one of the instructors know. 

```{r install, eval = FALSE}
install.packages("BiocManager")
BiocManager::install(c("scMerge", "devtools", 
                       "DropletUtils", "edgeR",
                       "ggpubr", "MAST", 
                       "plyr", "Rtsne", 
                       "scales", "scater", 
                       "scran", "tidyverse", 
                       "viridis"), version = "3.10")


devtools::install_github("SydneyBioX/scdney")
```

## For Windows users

If you could not run the last line of the above instructions, then please 

1. Try to install `Rtools35` from https://cran.r-project.org/bin/windows/Rtools/. 
1. Run `devtools::install_github("SydneyBioX/scdney")` again. 

## For Mac users

If you could not run the last line of the above instructions, then please 

1. Try to install `gfortran-6.1` from https://cran.r-project.org/bin/macosx/tools/.
1. Try the following code instead.

```{r mac install, eval = FALSE}
BiocManager::install(c("DescTools", "amap", 
                       "doParallel", "ggridges", 
                       "lme4", "mice", 
                       "methods", "caret", 
                       "randomForest", "clusteval", 
                       "dendextend", "gmodels", 
                       "e1071"), version = "3.10")

install.packages("https://github.com/SydneyBioX/scdney/releases/download/V0.1.4/scdney_0.1.4.tgz", repos = NULL)
```



# Clipboard

In some circumstances, we might need to share codes that are not currently a part of the existing materials. Please click [here](https://docs.google.com/document/d/136YffzxjkUM8RIGsnjD_yOkCTWw4Ijdwjy4_58nC-zc/edit?usp=sharing) to access these codes. 


# References

**Methodologies**

1. [scMerge leverages factor analysis, stable expression, and pseudoreplication to merge multiple single-cell RNA-seq datasets](http://www.pnas.org/lookup/doi/10.1073/pnas.1820006116), Proceedings of the National Academy of Sciences, United States of America, 2019. Yingxin Lin, Shila Ghazanfar, Kevin Y.X. Wang, Johann A. Gagnon-Bartsch, Kitty K. Lo, Xianbin Su, Ze-Guang Han, John T. Ormerod, Terence P. Speed, Pengyi Yang, Jean Y. H. Yang.

1. [Evaluating stably expressed genes in single cells](https://www.biorxiv.org/content/biorxiv/early/2018/11/22/229815.full.pdf) (Under review). Yingxin Lin, Shila Ghazanfar, Dario Strbenac, Andy Wang, Ellis Patrick, Dave Lin, Terence Speed, Jean Y. H. Yang, Pengyi Yang.

1. [Impact of similarity metrics on single-cell RNA-seq data clustering](https://doi.org/10.1093/bib/bby076), Briefings in Bioinformatics, 2018. Taiyun Kim, Irene Rui Chen, Yingxin Lin, Andy Y. Y. Wang, Jean Y. H. Yang, Pengyi Yang.


**Data**

1. [Single-cell RNA-Seq analysis reveals dynamic trajectories during mouse liver development](https://doi.org/10.1186/s12864-017-4342-x), BMC Genomics, 2017. Xianbin Su, Yi Shi, Xin Zou, Zhao-Ning Lu, Gangcai Xie, Jean Y. H. Yang, Chong-Chao Wu, Xiao-Fang Cui, Kun-Yan He, Qing Luo, Yu-Lan Qu, Na Wang, Lan Wang, Ze-Guang Han.

1. [A single‐cell transcriptomic analysis reveals precise pathways and regulatory mechanisms underlying hepatoblast differentiation](https://doi.org/10.1002/hep.29353), Hepatology, 2017. Li Yang, Wei‐Hua Wang, Wei‐Lin Qi, Zhen Guo, Erfei Bi, Cheng‐Ran Xu. 

```{r}
sessionInfo()
```

