---
title: "scClassify"
author: "Sydney Precision Bioinformatics Group"
date: "04 November 2019"
output:
  html_document:
    css: style.css
    code_folding: show
    fig_height: 12
    fig_width: 12
    toc: yes
    number_sections: true
    toc_depth: 3
    toc_float: yes
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(message = FALSE, 
                      warning = FALSE)
```


```{r}
library(SingleCellExperiment)
library(ggplot2)
library(Rtsne)
sce_liver <- readRDS("./data/liver_scMerge.rds")

ids <- colData(sce_liver)$batch %in% c("GSE87795", "GSE90047")
subset_data <- sce_liver[,ids]
```

# Extension: Can we use the data as reference for future cell type classification?


A common application of single-cell RNA sequencing (RNA-seq) data is to identify discrete cell types. To take advantage of the large collection of well-annotated scRNA-seq datasets, scClassify package implements a set of methods to perform accurate cell type classification based on ensemble learning and sample size calculation. 

The code below shows an example how to utilise a subset of data as reference data and then classify on the remain data.

```{r}
library(scClassify)

exprsMat_liver <- assay(subset_data, "scMerge")

set.seed(2019)

# subsample 80% of the data as training dataset, and rest as test
idx <- sample(ncol(exprsMat_liver), round(ncol(exprsMat_liver) * 0.8))

liver_exprsMat_train <- exprsMat_liver[, idx]
liver_exprsMat_test <- exprsMat_liver[, -idx]

liver_cellTypes_train <- subset_data$cellTypes[idx]
liver_cellTypes_test <- subset_data$cellTypes[-idx]


table(liver_cellTypes_train)
```

We first perform non-ensemble scClassify by using 80% data as our reference dataset. We use `WKNN` as the KNN algorithm, `DE` (differential expression genes) as the gene selection method, and lastly pearson as the similarity calculation method.


```{r}
trainLiver <- scClassify::train_scClassify(exprsMat_train = liver_exprsMat_train, 
                                           cellTypes_train = liver_cellTypes_train,
                                           algorithm = "WKNN",
                                           selectFeatures = c("limma"),
                                           similarity = c("pearson"),
                                           returnList = FALSE,
                                           verbose = TRUE)

trainLiver
```


We can visualise the cell type hirearchy tree using `plotCellTypeTree()`:

```{r}
scClassify::plotCellTypeTree(trainLiver@cellTypeTree)
```

Next, we perform `predict_scClassify()` with our trained model `trainRes = trainLiver` to predict the cell types of our query data matrix `liver_exprsMat_test`. Here, we used `pearson` as similarity metrics.

```{r}
pred_res <- scClassify::predict_scClassify(exprsMat_test = liver_exprsMat_test, 
                               cellTypes_test = liver_cellTypes_test,
                               trainRes = trainLiver,
                               algorithm = "WKNN",
                               features = c("limma"),
                               similarity = c("pearson"),
                               prob_threshold = 0.7,
                               verbose = TRUE)

scClassify_res <- pred_res$pearson_WKNN_limma$predRes
```


Check predicting results with the original labels.


```{r}
# create tsne object
set.seed(123)
tsne_result_test = Rtsne::Rtsne(t(liver_exprsMat_test), check_duplicates = FALSE)

#################################################

plot_data = data.frame(
  tsne1 = rep(tsne_result_test$Y[,1], 2),
  tsne2 = rep(tsne_result_test$Y[,2], 2),
  cluster = factor(c(liver_cellTypes_test, scClassify_res)),
  label = rep(c("Truth", "scClassify"), each = length(liver_cellTypes_test)))


ggplot(plot_data, aes(x = tsne1, y = tsne2, colour = cluster)) +
  geom_point(size = 2) +
  labs(title = "t-SNE plot") +
  facet_grid(~label) +
  theme(legend.position = "bottom")
```


```{r}
tab <- table(pred_res$pearson_WKNN_limma$predRes, liver_cellTypes_test)
knitr::kable(tab)
```




# Session Info
```{r}
sessionInfo()
```

